mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -744,17 +740,25 @@ impl Nexus {
     /// In case concurrent pause requests take place, the other callers
     /// will wait till the nexus is resumed and will continue execution
     /// with the nexus paused once they are awakened via resume().
-    /// Note: in order to handle cuncurrent pauses properly, this function must
+    /// Note: in order to handle concurrent pauses properly, this function must
     /// be called only from the master core.
-    pub(crate) async fn pause(&mut self) -> Result<(), Error> {
+    pub async fn pause(&mut self) -> Result<(), Error> {
         assert_eq!(Cores::current(), Cores::first());
 
-        match self.pause_state {
-            // Pause nexus if its unpaused.
-            NexusPauseState::Unpaused => {
-                self.pause_state = NexusPauseState::Pausing;
+        let state = self
+            .pause_state
+            .compare_exchange(
```

Comment:
Using atomic operations looks a bit overkillish here, as 'pause()' is executed always on a dedicated core (enforced by the assertion above, as we access nexus via a mutable reference).
Just a minor nit anyways, we can fix it later, as it is harmless.

Reply:
Yes agreed. However, the actual state itself might be consulted by the gRPC layer which lives on a different thread. We don't do that now, but we might need this in the future as we should not allow changes to a nexus that is pausing.